src/Testing/test/RetryTest.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Testing
+{
+    [Retry]
+    public class RetryTest
+    {
+        private static int _retryFailsUntil3 = 0;
+
+        [Fact]
+        public void RetryFailsUntil3()
+        {
+            _retryFailsUntil3++;
+            if (_retryFailsUntil3 != 2) throw new Exception("NOOOOOOOO");
+        }
+
+        private static int _canOverrideRetries = 0;
```

Comment:
I departed from the usual pattern here because these are doing a bad thing in that each test has its own static int, but I felt that was cleaner than adding some kind of internal context that we can look at since this is just test code.  So the grouping is really a static field per test, it would be really bad if the tests mucked with another one of these fields.